test/S3Test.js

```diff
@@ -1,285 +1,343 @@
-    })
-  })
+            client.putFile.getCall(0).args[1].should.equal('/node_js_logo.png')
 
-  it('should remove a file', function(done) {
-    var client = {
-      deleteFile: sinon.stub()
-    }
+            done()
+        })
+    })
 
-    var S3 = proxyquire('../lib/S3', {
-      'knox': {
-        createClient: function () {
-          return client
+    it('should remove a file', function (done) {
+        var client = {
+            deleteFile: sinon.stub()
         }
-      }
-    })
 
-    var s3 = new S3({
-      key: 'PUT_YOUR_KEY_HERE',
-      secret: 'PUT_YOUR_BUCKET_HERE',
-      bucket: 'PUT_YOUR_BUCKET_HERE',
-      region: 'PUT_YOUR_REGION_HERE'
+        var S3 = proxyquire('../lib/S3', {
+            'knox': {
+                createClient: function () {
+                    return client
+                }
+            }
+        })
+
+        var s3 = new S3({
+            key: 'PUT_YOUR_KEY_HERE',
+            secret: 'PUT_YOUR_BUCKET_HERE',
+            bucket: 'PUT_YOUR_BUCKET_HERE',
+            region: 'PUT_YOUR_REGION_HERE'
+        })
+
+        client.deleteFile.callsArg(1)
+
+        s3.remove({
+            url: 'foo'
+        }, function () {
+            client.deleteFile.getCall(0).args[0].should.equal('foo')
+
+            done()
+        })
     })
 
-    client.deleteFile.callsArg(1)
-
-    s3.remove({
-      url: 'foo'
-    }, function() {
-      client.deleteFile.getCall(0).args[0].should.equal('foo')
+    it('should not remove a file with no URL', function (done) {
+        var client = {
+            deleteFile: sinon.stub()
+        }
 
-      done()
+        var S3 = proxyquire('../lib/S3', {
+            'knox': {
+                createClient: function () {
+                    return client
+                }
+            }
+        })
+
+        var s3 = new S3({
+            key: 'PUT_YOUR_KEY_HERE',
+            secret: 'PUT_YOUR_BUCKET_HERE',
+            bucket: 'PUT_YOUR_BUCKET_HERE',
+            region: 'PUT_YOUR_REGION_HERE'
+        })
+
+        client.deleteFile.callsArg(1)
+
+        s3.remove({
+            url: null
+        }, function () {
+            client.deleteFile.callCount.should.equal(0)
+
+            done()
+        })
     })
-  })
-
-  it('should not remove a file with no URL', function(done) {
-    var client = {
-      deleteFile: sinon.stub()
-    }
 
-    var S3 = proxyquire('../lib/S3', {
-      'knox': {
-        createClient: function () {
-          return client
+    // Added by Alex Kruegger
+    it('should pass the tests for fixed bug', function (done) {
+        var client = {
+            _cb: sinon.stub()
         }
-      }
-    })
 
-    var s3 = new S3({
-      key: 'PUT_YOUR_KEY_HERE',
-      secret: 'PUT_YOUR_BUCKET_HERE',
-      bucket: 'PUT_YOUR_BUCKET_HERE',
-      region: 'PUT_YOUR_REGION_HERE'
+        var S3 = proxyquire('../lib/S3', {
+            'knox': {
+                createClient: function () {
+                    return client
```

Comment:
`client` has a stubbed `_cb` field - is this necessary?  Usually you'd do this because you want to stub some behaviour but you don't seem to do anything with `client._cb` after this.